File: Leetcode/python/sol2570.py
# https://leetcode.com/problems/merge-two-2d-arrays-by-summing-values/description/
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:

        seen = {}
        for i in nums1+nums2:
            seen[i[0]] = seen.get(i[0], 0) + i[1]

        result = [[i ,j] for i, j in seen.items()]
        result.sort(key=lambda x :x[0])

        return result

File: Leetcode/cpp/sol2570.cpp

// https://leetcode.com/problems/merge-two-2d-arrays-by-summing-values/description/

#include <algorithm> // For std::copy and std::move (C++11 and later) b

class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& nums1, vector<vector<int>>& nums2) {

        map<int,int> seen;
        for(auto i : nums2){
            nums1.push_back(i);
        }
        for (auto i: nums1){
            if(seen.find(i[0]) == seen.end()){
                seen[i[0]] = 0;
            }
            seen[i[0]] += i[1];
        }
        vector <vector<int>> result;
        for(auto i: seen){
            result.push_back({i.first, i.second});
        }

        return result;
    }
};